dkg/nodesigs.go

```diff
@@ -0,0 +1,100 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package dkg
+
+import (
+	"context"
+
+	k1 "github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/k1util"
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/cluster"
+	"github.com/obolnetwork/charon/dkg/bcast"
+	dkgpb "github.com/obolnetwork/charon/dkg/dkgpb/v1"
+)
+
+const nodeSigMsgID = "/charon/dkg/lock_hash_k1_sig"
+
+func nodeSigMsgIDs() []string {
+	return []string{nodeSigMsgID}
+}
+
+// nodeSigBcast handles broadcasting of K1 signatures over the lock hash via the bcast protocol.
+type nodeSigBcast struct {
+	otherSigs   chan *dkgpb.MsgLockHashK1Sig
+	bcastFunc   bcast.BroadcastFunc
+	operatorAmt int
+}
+
+// newNodeSigBcast returns a new instance of nodeSigBcast with the given operatorAmt.
+// It registers bcast handlers on bcastComp.
+func newNodeSigBcast(operatorAmt int, bcastComp *bcast.Component) nodeSigBcast {
+	// bcast returns len(peers)-1 messages to each peer, so that senders don't get their own message
```

Comment:
not sure this comment is correct, seems like it applies to `otherSigs` field rather than `ret` instance

Reply:
you actually spotted a bug, the channel must be `len(operators)-1`, not `len(operators)`: it slipped away during refactoring